### ISSUE TITLE
godco eoc error in 0.H stable

### ISSUE THREAD

**Issue by SamGondra** [CleverRaven/Cataclysm-DDA#78112]
### Describe the bug

eoc fired while I was in exodii base, changing terrain there instead of where it should.
sadly do not have a save from before it fired, so the savefile is from after, not sure if it gonna help.

### Attach save file

[Reform-trimmed.tar.gz](https://github.com/user-attachments/files/17892990/Reform-trimmed.tar.gz)

### Steps to reproduce

probably do the pebble pusher mission and then be far away from the compound several days later when the EOC fires?

### Expected behavior

the EOC firing where it is supposed to

### Screenshots

![Pretty sure this is not supposed to happen](https://github.com/user-attachments/assets/a63c0587-519e-4eca-813f-019f0dc2e758)

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Linuxmint; Description: Linux Mint 21.3; Release: 21.3; Codename: virginia; 
- Game Version: 78fd6e3 [64-bit]
- Graphics Version: Tiles
- Game Language: English <color_dark_gray>(100.0%)</color> [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Slots [cbm_slots]
]

### Additional context

[debug.log](https://github.com/user-attachments/files/17892969/debug.log)

**Comment by harakka** [comment]
Yeah this appears to be tied to completing MISSION_GODCO_KOSTAS_2 after which godco_place_herb_garden_new runs 72 hours  later. The eoc has  `{ "mapgen_update": "place_herb_garden", "om_terrain": "godco_1_2", "target_var": { "context_val": "search_locale" } },` As an aside I went looking for what `"context_val": "search_locale"` is supposed to do and I'm literally only seeing it in godco eocs, that's somewhat sus.

**Comment by Procyonae** [comment]
Nope I fixed this for exp in #77665, didn't realise the issue predated 0.H
